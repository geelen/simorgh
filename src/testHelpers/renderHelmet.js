/*
 * Similar to this problem https://github.com/testing-library/react-testing-library/issues/402
 * This helper was created to solve the problem of rendering helmet/head contents in snapshots.
 * Pass in a component that uses helmet somewhere in the component tree.
 * The full html tree is rendered and returned which you can then use to snapshot helmet/head contents.
 * Example of use in a test file:

 it('should render correctly', async () => {
    const html = await renderHelmet(<SomeComponent {...someProps} />);
    expect(html).toMatchSnapshot();
  });

 */

import { render, waitForDomChange } from '@testing-library/react';

export default async component => {
  render(component);

  await waitForDomChange();

  const htmlElement = document.querySelector('html');

  const helmetElements = document.querySelectorAll(
    '[data-react-helmet="true"]',
  );

  const removeHelmetAttributes = el => el.removeAttribute('data-react-helmet'); // remove react-helmet attribute noise from elements

  removeHelmetAttributes(htmlElement); // remove react-helmet attribute noise from elements

  Array.from(helmetElements).forEach(removeHelmetAttributes);

  return document.querySelector('html');
};
