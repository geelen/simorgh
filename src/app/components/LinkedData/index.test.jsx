import React from 'react';
import renderHelmet from '../../../testHelpers/renderHelmet';
import LinkData from '.';

describe('LinkData', () => {
  const props = {
    isAmp: false,
    brandName: 'BBC News',
    canonicalLink: 'https://www.bbc.com/news/articles/c9rpqy7pmypo',
    lang: 'en-GB',
    seoHeadline: 'Royal wedding 2018: Bouquet laid on tomb of unknown warrior',
    type: 'Article',
    lastUpdated: '2019-02-26T11:25:10.555Z',
    firstPublished: '2018-10-10T16:19:31.344Z',
    noBylinesPolicy: 'https://www.bbc.com/news/help-41670342#authorexpertise',
    publishingPrinciples: 'https://www.bbc.com/news/help-41670342',
    logoUrl:
      'https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png',
    about: [
      {
        '@type': 'Person',
        name: 'Duchess of Sussex',
        alternateName: 'Meghan Markle',
        sameAs: ['http://dbpedia.org/resource/Meghan_Markle'],
      },
    ],
  };

  const propsWithNoAbout = {
    isAmp: false,
    brandName: 'BBC News',
    canonicalLink: 'https://www.bbc.com/news/articles/c9rpqy7pmypo',
    lang: 'en-GB',
    seoHeadline: 'Royal wedding 2018: Bouquet laid on tomb of unknown warrior',
    type: 'Article',
    service: 'news',
    lastUpdated: '2019-02-26T11:25:10.555Z',
    firstPublished: '2018-10-10T16:19:31.344Z',
    noBylinesPolicy: 'https://www.bbc.com/news/help-41670342#authorexpertise',
    publishingPrinciples: 'https://www.bbc.com/news/help-41670342',
    logoUrl:
      'https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png',
  };

  const propsForFrontpage = {
    isAmp: false,
    brandName: 'BBC News',
    canonicalLink: 'https://www.bbc.com/igbo',
    lang: 'ig',
    seoHeadline: 'Ogbako',
    type: 'WebPage',
    service: 'igbo',
    lastUpdated: null,
    firstPublished: null,
    noBylinesPolicy: 'https://www.bbc.com/news/help-41670342#authorexpertise',
    publishingPrinciples: 'https://www.bbc.com/news/help-41670342',
    logoUrl: 'https://news.files.bbci.co.uk/ws/img/logos/og/igbo.png',
  };

  it('should correctly render metadata for links', async () => {
    const html = await renderHelmet(<LinkData {...props} />);
    expect(html).toMatchSnapshot();
  });

  it('should correctly render metadata with no about tags for links', async () => {
    const html = await renderHelmet(<LinkData {...propsWithNoAbout} />);
    expect(html).toMatchSnapshot();
  });

  it('should correctly render metadata for frontpages', async () => {
    const html = await renderHelmet(<LinkData {...propsForFrontpage} />);
    expect(html).toMatchSnapshot();
  });
});
