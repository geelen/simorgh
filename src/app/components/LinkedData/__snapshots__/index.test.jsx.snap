// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkData should correctly render metadata for frontpages 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"@context":"http://schema.org","@type":"Article","url":"https://www.bbc.com/news/articles/c9rpqy7pmypo","publisher":{"@type":"NewsMediaOrganization","name":"BBC News","publishingPrinciples":"https://www.bbc.com/news/help-41670342","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"}},"image":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"},"thumbnailUrl":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png","mainEntityOfPage":{"@type":"WebPage","@id":"https://www.bbc.com/news/articles/c9rpqy7pmypo","name":"Royal wedding 2018: Bouquet laid on tomb of unknown warrior"},"headline":"Royal wedding 2018: Bouquet laid on tomb of unknown warrior","datePublished":"2018-10-10T16:19:31.344Z","dateModified":"2019-02-26T11:25:10.555Z","author":{"@type":"NewsMediaOrganization","name":"BBC News","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"},"noBylinesPolicy":"https://www.bbc.com/news/help-41670342#authorexpertise"},"about":[{"@type":"Person","name":"Duchess of Sussex","alternateName":"Meghan Markle","sameAs":["http://dbpedia.org/resource/Meghan_Markle"]}]}
    </script>
    <script
      type="application/ld+json"
    >
      {"@context":"http://schema.org","@type":"Article","url":"https://www.bbc.com/news/articles/c9rpqy7pmypo","publisher":{"@type":"NewsMediaOrganization","name":"BBC News","publishingPrinciples":"https://www.bbc.com/news/help-41670342","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"}},"image":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"},"thumbnailUrl":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png","mainEntityOfPage":{"@type":"WebPage","@id":"https://www.bbc.com/news/articles/c9rpqy7pmypo","name":"Royal wedding 2018: Bouquet laid on tomb of unknown warrior"},"headline":"Royal wedding 2018: Bouquet laid on tomb of unknown warrior","datePublished":"2018-10-10T16:19:31.344Z","dateModified":"2019-02-26T11:25:10.555Z","author":{"@type":"NewsMediaOrganization","name":"BBC News","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"},"noBylinesPolicy":"https://www.bbc.com/news/help-41670342#authorexpertise"}}
    </script>
    <script
      type="application/ld+json"
    >
      {"@context":"http://schema.org","@type":"WebPage","url":"https://www.bbc.com/igbo","publisher":{"@type":"NewsMediaOrganization","name":"BBC News","publishingPrinciples":"https://www.bbc.com/news/help-41670342","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://news.files.bbci.co.uk/ws/img/logos/og/igbo.png"}},"image":{"@type":"ImageObject","width":1024,"height":576,"url":"https://news.files.bbci.co.uk/ws/img/logos/og/igbo.png"},"thumbnailUrl":"https://news.files.bbci.co.uk/ws/img/logos/og/igbo.png","mainEntityOfPage":{"@type":"WebPage","@id":"https://www.bbc.com/igbo","name":"Ogbako"}}
    </script>
  </head>
  <body>
    <div />
  </body>
</html>
`;

exports[`LinkData should correctly render metadata for links 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"@context":"http://schema.org","@type":"Article","url":"https://www.bbc.com/news/articles/c9rpqy7pmypo","publisher":{"@type":"NewsMediaOrganization","name":"BBC News","publishingPrinciples":"https://www.bbc.com/news/help-41670342","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"}},"image":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"},"thumbnailUrl":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png","mainEntityOfPage":{"@type":"WebPage","@id":"https://www.bbc.com/news/articles/c9rpqy7pmypo","name":"Royal wedding 2018: Bouquet laid on tomb of unknown warrior"},"headline":"Royal wedding 2018: Bouquet laid on tomb of unknown warrior","datePublished":"2018-10-10T16:19:31.344Z","dateModified":"2019-02-26T11:25:10.555Z","author":{"@type":"NewsMediaOrganization","name":"BBC News","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"},"noBylinesPolicy":"https://www.bbc.com/news/help-41670342#authorexpertise"},"about":[{"@type":"Person","name":"Duchess of Sussex","alternateName":"Meghan Markle","sameAs":["http://dbpedia.org/resource/Meghan_Markle"]}]}
    </script>
  </head>
  <body>
    <div />
  </body>
</html>
`;

exports[`LinkData should correctly render metadata with no about tags for links 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"@context":"http://schema.org","@type":"Article","url":"https://www.bbc.com/news/articles/c9rpqy7pmypo","publisher":{"@type":"NewsMediaOrganization","name":"BBC News","publishingPrinciples":"https://www.bbc.com/news/help-41670342","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"}},"image":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"},"thumbnailUrl":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png","mainEntityOfPage":{"@type":"WebPage","@id":"https://www.bbc.com/news/articles/c9rpqy7pmypo","name":"Royal wedding 2018: Bouquet laid on tomb of unknown warrior"},"headline":"Royal wedding 2018: Bouquet laid on tomb of unknown warrior","datePublished":"2018-10-10T16:19:31.344Z","dateModified":"2019-02-26T11:25:10.555Z","author":{"@type":"NewsMediaOrganization","name":"BBC News","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"},"noBylinesPolicy":"https://www.bbc.com/news/help-41670342#authorexpertise"},"about":[{"@type":"Person","name":"Duchess of Sussex","alternateName":"Meghan Markle","sameAs":["http://dbpedia.org/resource/Meghan_Markle"]}]}
    </script>
    <script
      type="application/ld+json"
    >
      {"@context":"http://schema.org","@type":"Article","url":"https://www.bbc.com/news/articles/c9rpqy7pmypo","publisher":{"@type":"NewsMediaOrganization","name":"BBC News","publishingPrinciples":"https://www.bbc.com/news/help-41670342","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"}},"image":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"},"thumbnailUrl":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png","mainEntityOfPage":{"@type":"WebPage","@id":"https://www.bbc.com/news/articles/c9rpqy7pmypo","name":"Royal wedding 2018: Bouquet laid on tomb of unknown warrior"},"headline":"Royal wedding 2018: Bouquet laid on tomb of unknown warrior","datePublished":"2018-10-10T16:19:31.344Z","dateModified":"2019-02-26T11:25:10.555Z","author":{"@type":"NewsMediaOrganization","name":"BBC News","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://www.bbc.com/news/special/2015/newsspec_10857/bbc_news_logo.png"},"noBylinesPolicy":"https://www.bbc.com/news/help-41670342#authorexpertise"}}
    </script>
  </head>
  <body>
    <div />
  </body>
</html>
`;
